krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -0,0 +1,238 @@
+	// SetCNIType sets the cniType in the spec
+	// returns an error if the value is an unknown type or when the
+	// set fails
+	SetCNIType(s string) error
+	// SetNetworkInstanceName sets the name of the networkInstance in the spec
+	// returns an error when the set fails
+	SetNetworkInstanceName(s string) error
+	// SetSpec sets the spec attributes in the kubeobject
+	SetSpec(*nephioreqv1alpha1.InterfaceSpec) error
+
+	DeleteAttachmentType() error
+
+	DeleteCNIType() error
+}
+
+// NewFromYAML creates a new parser interface
+// It expects a raw byte slice as input representing the serialized yaml file
+func NewFromYAML(b []byte) (Interface, error) {
+	o, err := fn.ParseKubeObject(b)
+	return &itfce{
+		o: o,
+	}, err
+}
+
+// NewFromGoStruct creates a new parser interface
+// It expects a interface go struct
+func NewFromGoStruct(x *nephioreqv1alpha1.Interface) (Interface, error) {
+	b, err := yaml.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+	return NewFromYAML(b)
+}
+
+type itfce struct {
+	o *fn.KubeObject
+}
+
+// GetKubeObject returns the present kubeObject
+func (r *itfce) GetKubeObject() *fn.KubeObject {
+	return r.o
+}
+
+func (r *itfce) GetGoStruct() (*nephioreqv1alpha1.Interface, error) {
+	x := &nephioreqv1alpha1.Interface{}
+	if err := yaml.Unmarshal([]byte(r.o.String()), x); err != nil {
+		return nil, err
+	}
+	return x, nil
+}
+
+func (r *itfce) GetAttachmentType() string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString("spec", "attachmentType")
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *itfce) GetCNIType() string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString("spec", "cniType")
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *itfce) GetNetworkInstanceName() string {
+	if r.o == nil {
+		return ""
+	}
+	s, ok, err := r.o.NestedString("spec", "networkInstance", "name")
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *itfce) SetAttachmentType(s string) error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	// validation -> should be part of the interface api repo
+	switch s {
+	case string(nephioreqv1alpha1.AttachmentTypeNone):
+	case string(nephioreqv1alpha1.AttachmentTypeVLAN):
+	default:
+		return fmt.Errorf("unknown attachmentType")
+	}
+
+	if err := r.o.SetNestedField(s, "spec", "attachmentType"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *itfce) DeleteAttachmentType() error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	_, err := r.o.RemoveNestedField("spec", "attachmentType")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *itfce) SetCNIType(s string) error {
+	if r.o == nil {
+		return fmt.Errorf("KubeObject not initialized")
+	}
+	// validation -> should be part of the interface api repo
+	switch s {
+	case string(nephioreqv1alpha1.CNITypeIPVLAN):
```

Comment:
It's tempting for me to store the supported CNI types in a slice (in a file where they are defined)  named SupportedCNIs and add a method  "IsCNISupported(att String) bool". That method will look up in the slice and return a boolean. That way we can capture this in a common way and future proof as well. All we need to do is update the slice when we add/delete support for an CNI type.